lib/_stream_readable.js

```diff
@@ -72,7 +72,7 @@ function ReadableState(options, stream) {
   this.highWaterMark = (hwm || hwm === 0) ? hwm : defaultHwm;
 
   // cast to ints.
-  this.highWaterMark = ~~this.highWaterMark;
+  this.highWaterMark = Math.floor(this.highWaterMark);
```

Comment:
Minor nit: This changes the result when passed a negative floating point number.